jsre/jsre.go

```diff
@@ -70,6 +73,28 @@ func New(assetPath string) *JSRE {
 	return re
 }
 
+// randomSource returns a pseudo random float64 in the range [0.0-1.0).
+func randomSource() float64 {
+	var src rand.Source
+	var once sync.Once
+
+	once.Do(func() {
+		bytes := make([]byte, 8)
+		seed := time.Now().UnixNano()
+		if _, err := crand.Read(bytes); err == nil {
+			seed = int64(binary.LittleEndian.Uint64(bytes))
+		}
+		src = rand.NewSource(seed)
+	})
```

Comment:
Also I don't think this `once` is correct. You create a new once every time, so every `Do` invocation will use a different underlying `sync.Once`, and hence each will actually run (https://golang.org/src/sync/once.go?s=1139:1166#L25).

I think the correct solution is not to use this function inside randomSource, but rather for this function to **return** a random source (i.e. return func() float64). Then you could do the random seed generation and assemble a function that will use it for everything later.

I.e. 

```go
func newRandomSource() func() float64 {
	bytes := make([]byte, 8)
	seed := time.Now().UnixNano()
	if _, err := crand.Read(bytes); err == nil {
		seed = int64(binary.LittleEndian.Uint64(bytes))
	}
	src = rand.NewSource(seed)
	return func() float64 {
		return float64(src.Int63()) / float64(1 << 64))
	}
}
```